components/merkledb/src/proof_map_index/proof.rs

```diff
@@ -270,9 +279,9 @@ pub struct MapProof<K, V> {
 /// See [`MapProof`] for an example of usage.
 ///
 /// [`MapProof`]: struct.MapProof.html#workflow
-#[derive(Debug, Serialize, Deserialize)]
-pub struct CheckedMapProof<K, V> {
-    entries: Vec<(K, Option<V>)>,
+#[derive(Debug, Serialize)]
+pub struct CheckedMapProof<'a, K, V> {
+    entries: &'a [OptionalEntry<K, V>],
```

Comment:
If borrowed data in checked proofs breaks Java binding (I vaguely remember something like that, but I *may* be mixing it up), then it's better to know now.